### ISSUE TITLE
m2m100 model do not support eval / predict on deepspeed + fp16 enviroment?

### ISSUE THREAD

**Issue by lavine-lmu** [huggingface/transformers#15570]
Hello everyone! Recently, I am finetuning m2m100 model, but find using ```deepspeed``` + ```fp16``` can caculate loss in train_step, but got ```nan``` loss in eval_step and predict_step. The ```run_translation.py``` is normally in ```mbart``` model and is also normarlly in ```m2m100``` model when do not using ```fp16```.  Has anyone encountered this issues?

**Comment by LysandreJik** [comment]
Maybe of interest to @patil-suraj (M2M100) and @stas00 (DeepSpeed)

**Comment by stas00** [comment]
I bet that if you remove deepspeed the problem won't go away, 

How was m2m100 pre-trained? if it's bf16 mixed precision (on TPUs) then it suffers the same problem as most other bf16-pretrained models (e.g. t5) - i.e. it can't be finetuned/eval'ed under fp16.

So if that's the case then you have 2 choices:

1. switch to fp32
2. switch to bf16 if you have access to Ampere GPUs (you'd need pytorch >= 1.9 for that) - you will get about the same performance as fp16.

**Comment by lavine-lmu** [comment]
@stas00 Thanks for your reply, I am switching to fp32, we can use deepspeed to finetune a ```m2m100``` model with deepspeed.

**Comment by patil-suraj** [comment]
> How was m2m100 pre-trained? 

@stas00 `M2M100` is trained on GPUs with `fp16`. Here is the original model https://github.com/pytorch/fairseq/tree/main/examples/m2m_100

**Comment by stas00** [comment]
So what happened:

If you look at the log, you can see it's trying to find the right loss scaling factor:
```
[2022-02-10 09:32:30,702] [INFO] [stage_1_and_2.py:1652:step] [deepscale] OVERFLOW! Rank 0 Skipping step. Attempted loss scale: 65536, reducing to 65536
  1%|▌                                                                                                                    | 1/189 [00:00<01:23,  2.26it/s][2022-02-10 09:32:31,160] [INFO] [stage_1_and_2.py:1652:step] [deepscale] OVERFLOW! Rank 0 Skipping step. Attempted loss scale: 65536, reducing to 32768.0
  2%|█▊                                                                                                                   | 3/189 [00:02<02:47,  1.11it/s][2022-02-10 09:32:33,056] [INFO] [stage_1_and_2.py:1652:step] [deepscale] OVERFLOW! Rank 0 Skipping step. Attempted loss scale: 32768.0, reducing to 16384.0
  2%|██▍                                                                                                                  | 4/189 [00:02<02:14,  1.37it/s][2022-02-10 09:32:33,518] [INFO] [stage_1_and_2.py:1652:step] [deepscale] OVERFLOW! Rank 0 Skipping step. Attempted loss scale: 16384.0, reducing to 8192.0
  3%|███                                                            
[...]  
   17%|███████████████████▋                                                                                                | 32/189 [00:16<01:12,  2.18it/s][2022-02-10 09:32:47,241] [INFO] [stage_1_and_2.py:1652:step] [deepscale] OVERFLOW! Rank 0 Skipping step. Attempted loss scale: 1, reducing to 1
 17%|████████████████████▎                                                                                               | 33/189 [00:16<01:12,  2.16it/s][2022-02-10 09:32:47,718] [INFO] [stage_1_and_2.py:1652:step] [deepscale] OVERFLOW! Rank 0 Skipping step. Attempted loss scale: 1, reducing to 1
 18%|████████████████████▊                                                                                               | 34/189 [00:17<01:12,  2.14it/s][2022-02-10 09:32:48,157] [INFO] [stage_1_and_2.py:1652:step] [deepscale] OVERFLOW! Rank 0 Skipping step. Attempted loss scale: 1, reducing to 1
  
```
and it fails to do so.

So in this particular case, configuring deepspeed.fp16 settings to start from a higher initial scale power overcomes the problem.

Please let me know if this resolves this issue for you, @lavine-lmu

**Comment by lavine-lmu** [comment]
it works after increase the ```initial_scale_power ``` from 16 to 32. Thank you very much! @stas00

**Comment by stas00** [comment]
also added this use case to the troubleshooting section here: https://github.com/huggingface/transformers/pull/15585

**Comment by joao-alves97** [comment]
@lavine-lmu were you able to achieve good results when fine-tuning M2M100? I'm trying to do that but unfortunately when I try to do that the loss decreases but the performance does not improve (it goes from around 30BLEU points without fine-tuning to 0.1 BLEU points after fine-tuning)
